### DISCUSSION TITLE
Propagating internal error message as API Errors

### DISCUSSION THREAD

**Discussion by AliSajid** [pocketbase/pocketbase#1094] [Q&A]
Hi,

I have Pocketbase set up the way I want. However, when using the API, I am seeing non-descriptive errors (e.g "Something went wrong". I can see the exact error in the log console with the failed API request. Is there an easy way to propagate that error message outside to the API?

**Comment by ganigeorgiev** [comment]
No, this is done for security to avoid accidentally leaking sensitive information. Only the "public-safe" serializable errors (eg. validation errors) will be returned under the `data` response key.

If you use PocketBase as framework you can overwrite/extend the global API error handler by doing something like this:

```go
app.OnBeforeServe().Add(func(e *core.ServeEvent) error {
	oldErrHandler := e.Router.HTTPErrorHandler
	e.Router.HTTPErrorHandler = func(c echo.Context, err error) {
		if c.Response().Committed {
			return
		}

		// do something...

		oldErrHandler(c, err)
	}

	return nil
})
```
_(there will be a hook in the near future to  make it a little bit easier)_
